vsintegration/src/FSharp.Editor/CodeFixes/AddTypeAnnotationToObjectOfIndeterminateType.fs

```diff
@@ -24,88 +23,98 @@ type internal AddTypeAnnotationToObjectOfIndeterminateTypeFixProvider [<Importin
-                        let hasRightParen = rightLoop sourceText.[declSpan.End] declSpan.End
-                        hasLeftParen && hasRightParen
-
-                    let changes =
-                        [
-                            if alreadyWrappedInParens then
-                                TextChange(TextSpan(declSpan.End, 0), ": " + typeString)
-                            else
-                                TextChange(TextSpan(declSpan.Start, 0), "(")
-                                TextChange(TextSpan(declSpan.End + 1, 0), ": " + typeString + ")")
-                        ]
-
-                    context.RegisterFsharpFix(CodeFix.AddTypeAnnotationToObjectOfIndeterminateType, title, changes)
-
-                | _ -> ()
-            | _ -> ()
-        }
-        |> Async.Ignore
-        |> RoslynHelpers.StartAsyncUnitAsTask(context.CancellationToken)
+                match lexerSymbolOpt with
+                | None -> return ValueNone
+                | Some lexerSymbol ->
+                    let! sourceText = context.GetSourceTextAsync()
+                    let textLine = sourceText.Lines.GetLineFromPosition position
+                    let textLinePos = sourceText.Lines.GetLinePosition position
+                    let fcsTextLineNumber = Line.fromZ textLinePos.Line
+
+                    let! _, checkFileResults =
+                        document.GetFSharpParseAndCheckResultsAsync(nameof AddTypeAnnotationToObjectOfIndeterminateTypeFixProvider)
+
+                    let decl =
+                        checkFileResults.GetDeclarationLocation(
+                            fcsTextLineNumber,
+                            lexerSymbol.Ident.idRange.EndColumn,
+                            textLine.ToString(),
+                            lexerSymbol.FullIsland,
+                            false
+                        )
+
+                    match decl with
+                    | FindDeclResult.DeclFound declRange when declRange.FileName = document.FilePath ->
+                        let declSpan = RoslynHelpers.FSharpRangeToTextSpan(sourceText, declRange)
+                        let declTextLine = sourceText.Lines.GetLineFromPosition declSpan.Start
+
+                        let symbolUseOpt =
+                            checkFileResults.GetSymbolUseAtLocation(
+                                declRange.StartLine,
+                                declRange.EndColumn,
+                                declTextLine.ToString(),
+                                lexerSymbol.FullIsland
+                            )
+
+                        match symbolUseOpt with
+                        | None -> return ValueNone
+                        | Some symbolUse ->
+                            match symbolUse.Symbol with
+                            | :? FSharpMemberOrFunctionOrValue as mfv when not mfv.FullType.IsGenericParameter ->
```

Comment:
This fixes https://github.com/dotnet/fsharp/issues/15811. Should be sufficient, I guess?